### PR TITLE
Process implementation

### PR DESCRIPTION
**Pull request by ziyue-pan**
## What we have done

- [x] coarse implementation of the handlers

## TODO (Maybe in next PR)

- [ ] authentication for restricted handlers
- [ ] filter the unwanted attributes in returned models
- [ ] test on these handlers
- [ ] some initial data

### CODE REVIEW

**Comment by AsterNighT** [comment]
Is is done now? If it is not ready yet, you can add a [WIP] in the title

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,7 +30,9 @@ jobs:
       - name: checkout
         uses: actions/checkout@v2
       - name: test
-        run: go test ./...
+        run: |
```
Comment (AsterNighT) [inline .github/workflows/ci.yml]: don't put it here. Update go.mod and go.sum instead

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,6 +46,7 @@ jobs:
         uses: actions/checkout@v2
       - name: test
         run: | 
+          go get github.com/labstack/gommon/log
```
Comment (AsterNighT) [inline .github/workflows/ci.yml]: ditto

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -57,6 +60,7 @@ jobs:
         uses: actions/checkout@v2
       - name: build
         run: |
+          go get github.com/labstack/gommon/log
```
Comment (AsterNighT) [inline .github/workflows/ci.yml]: ditto

**Review by AsterNighT — CHANGES_REQUESTED** [review]
Pls don't merge master. Use rebase instread. Your file changes are a mess now.

**Review by RupertPaoZ — COMMENTED** [review]
please use rebase

**Comment by AsterNighT** [comment]
If your are not sure how to make up this mess, you could create a new branch and cherrypick your changes, that'll be easier.

**Comment by ziyue-pan** [comment]
> please use rebase

> If your are not sure how to make up this mess, you could create a new branch and cherrypick your changes, that'll be easier.


Cleared the messy changes.

File: pkg/process/handler.go [hunk 4]
```diff
@@ -152,9 +254,23 @@ func (h *ProcessHandler) CreateMileStoneByDoctor(c echo.Context) error {
...
+
+	var Milestone MileStone
+	var err error
+	var Checked bool
+
+	DB.First(&Milestone, c.QueryParam("MileStoneID"))
+	Checked, err = strconv.ParseBool(c.Param("Checked"))
+
+	if err != nil {
+		log.Error("UpdateMileStoneByDoctor failed due to", err)
```
Comment (AsterNighT) [inline pkg/process/handler.go]: umm, why not use `c.Logger` as it is done elsewhere?

File: pkg/process/handler.go [hunk 5]
```diff
@@ -2,11 +2,17 @@ package process
...
+	"strconv"
+	"time"
 
 	"github.com/AsterNighT/software-engineering-backend/api"
+	"github.com/AsterNighT/software-engineering-backend/pkg/utils"
 	"github.com/labstack/echo/v4"
+	"github.com/labstack/gommon/log"
 )
 
+
```
Comment (AsterNighT) [inline pkg/process/handler.go]: And pls fmt your file

**Review by AsterNighT — CHANGES_REQUESTED** [review]

File: pkg/process/handler.go [hunk 6]
```diff
@@ -133,11 +217,25 @@ func (h *ProcessHandler) UpdateRegistrationStatus(c echo.Context) error {
...
 // @Produce json
-// @Success 200 {string} api.ReturnedData{}
+// @Success 204 {string} api.ReturnedData{}
 // @Router /milestone [POST]
 func (h *ProcessHandler) CreateMileStoneByDoctor(c echo.Context) error {
-	//verify identity
-	c.Logger().Debug("hello world")
+	DB := utils.GetDB()
+
+	RegistrationID, err := strconv.ParseUint(c.QueryParam("RegistrationID"), 10, 64)
```
Comment (AsterNighT) [inline pkg/process/handler.go]: why is this `RegistrationID` UpperCamalCase?

File: pkg/process/handler.go [hunk 7]
```diff
@@ -152,9 +250,23 @@ func (h *ProcessHandler) CreateMileStoneByDoctor(c echo.Context) error {
 // @Success 200 {string} api.ReturnedData{}
 // @Router /milestone/{MileStoneID} [PUT]
 func (h *ProcessHandler) UpdateMileStoneByDoctor(c echo.Context) error {
-	//verify identity
-	c.Logger().Debug("hello world")
-	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+	DB := utils.GetDB()
+
+	var Milestone MileStone
```
Comment (AsterNighT) [inline pkg/process/handler.go]: ditto

File: pkg/process/handler.go [hunk 8]
```diff
@@ -152,9 +250,23 @@ func (h *ProcessHandler) CreateMileStoneByDoctor(c echo.Context) error {
...
+	var err error
+	var Checked bool
```
Comment (AsterNighT) [inline pkg/process/handler.go]: ditto

Comment (AsterNighT) [reply pkg/process/handler.go]: And else

**Review by AsterNighT — CHANGES_REQUESTED** [review]

**Comment by ziyue-pan** [comment]
sovled code style

File: pkg/process/handler.go [hunk 9]
```diff
@@ -169,42 +171,42 @@ func (h *ProcessHandler) GetRegistrationByDoctor(c echo.Context) error {
...
 		return c.JSON(http.StatusBadRequest, api.Return("error", err))
 	}
 
 	// TODO validation for status
-	Status := RegistrationStatusEnum(c.QueryParam("Status"))
-	TerminatedCause := c.QueryParam("TerminatedCause")
+	status := RegistrationStatusEnum(c.QueryParam("status"))
+	terminatedCause := c.QueryParam("terminatedCause")
 
 	var Registration Registration
```
Comment (AsterNighT) [inline pkg/process/handler.go]:
```suggestion
	var registration Registration
```

**Review by AsterNighT — CHANGES_REQUESTED** [review]

File: pkg/process/handler.go (hunk 9 above)
Comment (ziyue-pan) [reply pkg/process/handler.go]: It is fixed

**Review by AsterNighT — APPROVED** [review]

**Comment by AsterNighT** [comment]
WAM, so what's the usage of `RegistrationStatusEnum` in `pkg/process/model.go`? Is it useless?

**Comment by ziyue-pan** [comment]
> WAM, so what's the usage of `RegistrationStatusEnum` in `pkg/process/model.go`? Is it useless?

We have to control the status to limited states. Maybe used in `validator`.

**Comment by AsterNighT** [comment]
ok

**Pull request merged**